shell/platform/darwin/common/framework/Source/FlutterCodecs.mm

```diff
@@ -31,10 +32,12 @@ + (instancetype)sharedInstance {
   return _sharedInstance;
 }
 
-- (NSData*)encode:(NSString*)message {
+- (NSData*)encode:(id)message {
+  NSAssert(!message || [message isKindOfClass:[NSString class]], @"");
+  NSString* stringMessage = message;
   if (message == nil)
     return nil;
```

Comment:
Consider moving this up a line right below the assert. The compiler should optimise out the 'copy' regardless, but it gets rid of use of `message` sooner in the code..